packages/backend/src/payment-method/local/service.ts

```diff
@@ -0,0 +1,289 @@
+      description: 'receive amount of local quote is non-positive',
+      retryable: false,
+      code: PaymentMethodHandlerErrorCode.QuoteNonPositiveReceiveAmount
+    })
+  }
+
+  return {
+    receiver: options.receiver,
+    walletAddress: options.walletAddress,
+    estimatedExchangeRate:
+      Number(receiveAmountValue) / Number(debitAmountValue),
+    debitAmount: {
+      value: debitAmountValue,
+      assetCode: options.walletAddress.asset.code,
+      assetScale: options.walletAddress.asset.scale
+    },
+    receiveAmount: {
+      value: receiveAmountValue,
+      assetCode: options.receiver.assetCode,
+      assetScale: options.receiver.assetScale
+    },
+    additionalFields: {}
+  }
+}
+
+async function pay(
+  deps: ServiceDependencies,
+  options: PayOptions
+): Promise<void> {
+  const { outgoingPayment, receiver, finalReceiveAmount, finalDebitAmount } =
+    options
+
+  const incomingPaymentId = receiver.incomingPayment.id.split('/').pop()
+  if (!incomingPaymentId) {
+    throw new PaymentMethodHandlerError('Received error during local payment', {
+      description: 'Failed to parse incoming payment on receiver',
+      retryable: false
+    })
+  }
+  const incomingPayment = await deps.incomingPaymentService.get({
+    id: incomingPaymentId
+  })
+  if (!incomingPayment) {
+    throw new PaymentMethodHandlerError('Received error during local payment', {
+      description: 'Incoming payment not found from receiver',
+      retryable: false
+    })
+  }
+
+  // TODO: remove incoming state check? perhaps only applies ilp account middleware where its checking many different things
+  if (incomingPayment.state === IncomingPaymentState.Pending) {
+    try {
+      await deps.accountingService.createLiquidityAccount(
+        incomingPayment,
+        LiquidityAccountType.INCOMING
+      )
+    } catch (err) {
+      if (!(err instanceof AccountAlreadyExistsError)) {
+        deps.logger.error(
+          { incomingPayment, err },
+          'Failed to create liquidity account for local incoming payment'
+        )
+        throw new PaymentMethodHandlerError(
+          'Received error during local payment',
+          {
+            description:
+              'Unknown error while trying to create liquidity account',
+            retryable: false
+          }
+        )
+      }
+    }
+  }
+
+  const transferOptions: TransferOptions = {
+    sourceAccount: outgoingPayment,
+    destinationAccount: incomingPayment,
+    sourceAmount: finalDebitAmount,
+    destinationAmount: finalReceiveAmount,
+    transferType: TransferType.TRANSFER,
+    // TODO: no timeout? theoretically possible (perhaps even correct?) but need to
+    // update IncomingPayment state to COMPLETE some other way. trxOrError.post usually
+    // will (via onCredit) but post will return error instead because its already posted.
+    timeout: deps.config.tigerBeetleTwoPhaseTimeout
```

Comment:
@mkurapov originally you suggested not passing in a timeout which seems like it could make sense, but in that case the incoming payment never gets completed. included the timeout to ensure it does. How important is not setting a timeout? I can eitehr address it here, make a new issue, or remove the todo if its no big deal.

Reply:
cleaned up the todo, linking to the single phase transfer issue